### PR TITLE
Bug 1800315: [release-4.3] Drop openshift/cluster-api in favour of openshift/machine-api-operator to get controller's changes

### PR DESCRIPTION
**Pull request by alexander-demichev**
Backport of https://github.com/openshift/cluster-api-provider-aws/pull/276
This drops openshift/cluster-api and in favour of vendoring from machine-api-operator to get the machine API types and controllers.

This is to bring the machine controller changes introduced by
https://github.com/openshift/machine-api-operator/pull/490
https://github.com/openshift/machine-api-operator/pull/488

which fixes
https://bugzilla.redhat.com/show_bug.cgi?id=1800315 and https://bugzilla.redhat.com/show_bug.cgi?id=1803762 

As a consequence this PR also drops cluster struct references to satisfy actuator interface.

### CODE REVIEW

File: pkg/actuators/machine/actuator_test.go [hunk 1]
```diff
@@ -78,85 +76,67 @@ func TestMachineEvents(t *testing.T) {
...
-				actuator.DeleteMachine(cluster, machine)
-			},
-			event: "Normal Deleted Deleted machine aws-actuator-testing-machine",
-		},
-		{
-			name:                    "Delete machine event succeed with stopped instances",
-			machine:                 machine,
-			describeInstancesOutput: stubDescribeInstancesOutput("ami-a9acbbd6", "i-02fcb933c5da7085c", ec2.InstanceStateNameStopped),
-			operation: func(actuator *Actuator, cluster *clusterv1.Cluster, machine *machinev1.Machine) {
-				actuator.DeleteMachine(cluster, machine)
```
Comment (JoelSpeed) [inline pkg/actuators/machine/actuator_test.go]: Are these tests no longer relevant?

Comment (alexander-demichev) [reply pkg/actuators/machine/actuator_test.go]: Thanks for noticing it

**Comment by enxebre** [comment]
cc @knobunc

**Pull request merged**